### ISSUE TITLE
Excess memory usage from publish - is this normal?

### ISSUE THREAD

**Issue by droidenator** [arobson/rabbot#6]
I have a worker that's publishing about 125 messages per second. Memory keeps growing from like 10MB up to over 1GB in the span of twenty minutes. 

The worker is pretty straight-forward. It simply pulls handles a message from a queue, maps some data, assigns it a routing key, and then publishes it to another queue. The queue the message is being published to is noAck. 

We've seen this issue occur in two of our workers and we able to fix the memory issue in one of them by replacing `publish` with raw AMQP lib; I haven't tried it in the other worker. My co-worker thinks that rabbot is storing the published messages in memory and not clearing them out fast enough but 125 messages/sec shouldn't be an excessive load, should it?

I'm trying to use memwatch-next to profile the worker and get a feel for where the issue is coming from. This seems like there's a memory leak somewhere or is this kind of expected given the volume of messages I'm getting?

**Comment by droidenator** [comment]
After some more testing, it appears there's a nasty memory leak in `publish`. I have a worker that ends up hitting around 1GB of memory after running for 20 minutes. I simply swapped rabbot out with wascally and now my worker runs around 100MB give or take. Unless something crazy was added to publish in rabbot, it seems like there's a big memory leak there.

I'm testing this with about 266 messages per second.
